code/modules/power/supermatter/supermatter.dm

```diff
@@ -121,6 +121,7 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	var/atom/movable/distortion_effect/distort
 
 	var/last_status
+	var/current_status
```

Comment:
You don't need to store this here, since outside of the update function it will always be the same as last_status. You can simply check last_status instead and keep current_status local

Reply:
fixed it